### PR TITLE
Localise dates shown in UI using QTimeZone

### PR DESCRIPTION
**Pull request by philmcmahon**
# Description
Currently dates are shown in securedrop-client in UTC. This change localises them. 

The motivation for this came out of working on https://github.com/freedomofpress/securedrop-client/issues/1563 - after adding timestamps I noticed that they were in UTC. This change doesn't add timestamps - it keeps the existing date format, just localised. 

I've just added a single unit test for this the 'datetime to MMM DD' bit - happy to add some tests for the other functions too if desirable (I'd either need to mock out QTimeZone or pass in timezone as an optional argument)

# Test Plan
I tested these changes on my qubes workstation by manually changing the timezone of `sd-app`:

`sudo timedatectl set-timezone 'Pacific/Auckland'`

These are the dates in the UI I expect to update:

Screenshot 2023-02-13 at 16 08 30

# Checklist

If these changes modify code paths involving cryptography, the opening of files in VMs or network (via the RPC service) traffic, Qubes testing in the staging environment is required. For fine tuning of the graphical user interface, testing in any environment in Qubes is required. Please check as applicable:

 - [x] I have tested these changes in the appropriate Qubes environment
 - [ ] I do not have an appropriate Qubes OS workstation set up (the reviewer will need to test these changes)
 - [ ] These changes should not need testing in Qubes

If these changes add or remove files other than client code, the AppArmor profile may need to be updated. Please check as applicable:

 - [ ] I have updated the [AppArmor profile](https://github.com/freedomofpress/securedrop-client/blob/HEAD/files/usr.bin.securedrop-client)
 - [x] No update to the AppArmor profile is required for these changes
 - [ ] I don't know and would appreciate guidance

If these changes modify the database schema, you should include a database migration. Please check as applicable:

 - [ ] I have written a migration and upgraded a test database based on `main` and confirmed that the migration is [self-contained] and applies cleanly
 - [ ] I have written a migration but have not upgraded a test database based on `main` and would like the reviewer to do so
 - [ ] I need help writing a database migration
 - [x] No database schema changes are needed

[self-contained]: https://github.com/freedomofpress/securedrop-client#generating-and-running-database-migrations

### CODE REVIEW

File: tests/gui/test_widgets.py [hunk 1]
```diff
@@ -18,6 +17,7 @@
 
 from securedrop_client import db, logic, storage
 from securedrop_client.app import threads
+from securedrop_client.gui.format_helpers import format_datetime_local
```
Comment (cfm) [inline tests/gui/test_widgets.py:20]: Per <https://app.circleci.com/pipelines/github/freedomofpress/securedrop-client/3200/workflows/379f7720-ef14-4b46-ac7b-bc95572cff16/jobs/21202/tests>, this should be:

```suggestion
from securedrop_client.gui.datetime_helpers import format_datetime_local
```

**Review by cfm — CHANGES_REQUESTED** [review]
Thanks, @philmcmahon!  I'm noting that this is not full internationalization for #1536, just time-zone localization.  With that in mind, I've tested this and can confirm that it works as expected.  I'm requesting one change to fix a failing test.

File: tests/gui/test_widgets.py (hunk 1 above)
Comment (philmcmahon) [reply tests/gui/test_widgets.py]: Oops! Thanks for this, last minute renaming

**Comment by philmcmahon** [comment]
Thanks @cfm - I'd not spotted https://github.com/freedomofpress/securedrop-client/issues/1536 - happy to take a look at that next.

**Review by cfm — CHANGES_REQUESTED** [review]
Thanks, @philmcmahon!  I didn't catch this yesterday because of the failing test, but you'll also need to:
1. rebase from `main`, which I've proposed in 2ccd3ce1c5ba62e8eba8c312ff52348578d2a480; and
2. fix the linter error in <https://app.circleci.com/pipelines/github/freedomofpress/securedrop-client/3207/workflows/11de510f-6ccf-4bc1-b2c8-78253be749b8/jobs/21325?invite=true#step-103-344>.

**Comment by philmcmahon** [comment]
Thanks @cfm - have pushed those changes

**Review by cfm — CHANGES_REQUESTED** [review]
Thanks, @philmcmahon.  Looks good substantively, but still failing linting per <https://app.circleci.com/pipelines/github/freedomofpress/securedrop-client/3216/workflows/a27243e5-372b-418f-b46c-10edb302addb/jobs/21490>.  Locally you can check and apply these linters with (in this case) `make check-black` and `make black`, respectively.  :-)

**Comment by philmcmahon** [comment]
Sorry, I could have discovered that myself! I'd been ignoring `make check` as it was flagging up formatting issues in files I hadn't touched. Think it's in a better state now but I'm still getting:

```
securedrop_client/logic.py:27:1: F401 'typing.Dict' imported but unused
securedrop_client/logic.py:27:1: F401 'typing.List' imported but unused
make: *** [lint] Error 1
```

when I look in `logic.py` there are instances of both Dict and List so wasn't sure what to do there

**Comment by cfm** [comment]
Thanks for following up on this pull request, @philmcmahon!  I'll keep an eye on your changes here until they pass linting.  :-)

**Comment by philmcmahon** [comment]
Sorry for the loong delay on this! Got the linter working on a linux machine and the tests working on my mac so I've a bit more confidence this will pass ci! I've rebased against main too.

**Comment by gonzalo-bulnes** [comment]
@cfm and I have identified some formatting changes that:

- are unrelated to localization
- shouldn't be necessary to clear the linting checks

My intention was to push a commit reverting those, but it doesn't seem like I can push to either `localise-dates` on the `guardian` remote. (Truth is I know there is a setting to allow maintainers to push to a branch when a PR is open, but I don't know how to see if it is enabled or how I am supposed to push. I'm asking around see if anyone knows more than I do.)

**Comment by gonzalo-bulnes** [comment]
**Further context**: I am looking into removing the unnecessary formatting changes so that we can merge this PR today and include it in the upcoming release :slightly_smiling_face: 

I'll duplicate the PR for that effect, I hope that's not too much trouble for you @philmcmahon. (I don't think it should, if I'm wrong, feel free to ping me for cleanup/help @philmcmahon!)

**Pull request closed without merging**